openedx/core/djangoapps/content_libraries/libraries_index.py

```diff
@@ -242,17 +243,21 @@ def get_item_definition(cls, item):
 
 @receiver(CONTENT_LIBRARY_CREATED)
 @receiver(CONTENT_LIBRARY_UPDATED)
-@receiver(LIBRARY_BLOCK_CREATED)
-@receiver(LIBRARY_BLOCK_UPDATED)
-@receiver(LIBRARY_BLOCK_DELETED)
-def index_library(sender, library_key, **kwargs):  # pylint: disable=unused-argument
+def index_library(**kwargs):
     """
     Index library when created or updated, or when its blocks are modified.
     """
+    content_library = kwargs.get('content_library', None)
+    if not content_library or not isinstance(content_library, ContentLibraryData):
+        log.error('Received null or incorrect data for event')
+        return
```

Comment:
Just require this as a typed argument, it's clearer.

Please do the same everywhere else.

```suggestion
def index_library(sender, content_library: ContentLibraryData, **kwargs):  # pylint: disable=unused-argument
    """
    Index library when created or updated, or when its blocks are modified.
    """
```

Reply:
@pomegranited I suggested @rpenido to use `**kwargs` as documented in [openedx_events docs](https://github.com/openedx/openedx-events/blob/c0eb4ba1a3d7d066d58e5c87920b8ccb0645f769/docs/how-tos/using-events.rst?plain=1#L25-L29) and as seen in other examples like [CATALOG_CHANGE handler example](https://github.com/openedx/course-discovery/blob/a948a741b1839e845e07c762bab6ee51622daea5/course_discovery/apps/course_metadata/signals.py#L334-L346). I think the reason for them to be handled like this is to avoid raising exceptions and stop processing of all handlers in case of one handler receiving incorrect input. This is just another level of safety as openedx_event does [validate the input data](https://github.com/openedx/openedx-events/blob/c0eb4ba1a3d7d066d58e5c87920b8ccb0645f769/openedx_events/tooling.py#L125-L152) while producing.

But this is probably more useful in case of receiving events from event bus and not just as django signals.

Reply:
Oh! Ok, I think it looks awful, but if that's the way it's supposed to be, then I retract my comment :)

> But this is probably more useful in case of receiving events from event bus and not just as django signals.

Sure, but since these events may end up on the bus someday, it's better to code them the right way now.

Reply:
> Ok, I think it looks awful,

:sweat_smile: Yep.